### PR TITLE
KAFKA-14133: Migrate ChangeLogReader mock in TaskManagerTest to Mockito

### PR DESCRIPTION
**Pull request by clolov**
This pull request takes a similar approach as the one outlined in https://github.com/apache/kafka/pull/13529 to move each mock separately to ease reviewing the code.

Once https://github.com/apache/kafka/pull/13529 is merged this pull request will be rebased on top (as some of the changes are present in both pull requests to make the tests pass).

### CODE REVIEW

**Comment by clolov** [comment]
Heya @cadonna! I hope this attempt is what you had in mind? Unless I am wrong the removals detailed in the pull request are sensible as Mockito should be returning empty collections for method invocations on mocks based on "By default, for all methods that return a value, a mock will return either null, a primitive/primitive wrapper value, or an empty collection, as appropriate" (source: https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#stubbing)

**Comment by clolov** [comment]
I believe the related test failures are due to the same problem as https://github.com/apache/kafka/pull/13529#discussion_r1168432918. Once that PR is merged and this one is rebased those tests should pass.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java]: This line can also be removed.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java]: This line can also be removed.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:3025]: Since the mock was reset to strict, I think you need to verify or explicitly stub the call to `changeLogReader.completedChangelogs()`. Otherwise, the test would not fail, if `completedChangelogs()` were not called.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:3097]: Same here.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:3148]: Same here

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:3277]: Same here

**Review by cadonna — COMMENTED** [review]
Thanks for the PR, @clolov !

Here my feedback!

Let's remember to not merge this PR before https://github.com/apache/kafka/pull/13529 is merged and this PR rebased on trunk.

**Comment by clolov** [comment]
Great! Thank you @cadonna, I will aim to address the comments either today or tomorrow!

**Comment by cadonna** [comment]
@clolov Thanks for the updates! I just merged https://github.com/apache/kafka/pull/13529. Could you please rebase this PR?

**Review by cadonna — APPROVED** [review]
Thanks @clolov !

LGTM!

I restarted the builds. Once the builds are acceptable, I am going to merge this PR.

**Pull request merged**
